metrics-dropwizard/src/main/scala/com/mesosphere/usi/metrics/dropwizard/conf/MetricsSettings.scala

```diff
@@ -0,0 +1,100 @@
+package com.mesosphere.usi.metrics.dropwizard.conf
+
+import com.typesafe.config.Config
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.duration._
+import net.ceedubs.ficus.Ficus.{finiteDurationReader, toFicusConfig}
+
+case class HistorgramSettings(reservoirHighestTrackableValue: Long = 3600000000000L,
+                              reservoirSignificantDigits: Int = 3,
+                              reservoirResetPeriodically: Boolean = true,
+                              reservoirResettingInterval: FiniteDuration = 5.seconds,
+                              reservoirResettingChunks: Int = 0
+                             ) {
+  require(reservoirHighestTrackableValue > 1L, "reservoir-highest-trackable-value: should be > 1")
+  require(reservoirSignificantDigits >= 0 && reservoirSignificantDigits <= 5, "reservoir-significant-digits should be >= 0 and <= 5")
+  require(reservoirResettingInterval > Duration.Zero, "reservoir-resetting-interval should be > 0")
+  require(reservoirResettingChunks == 0 || reservoirResettingChunks >= 2, "reservoir-resetting-chunks should be == 0 or >= 2")
+}
+
+object HistorgramSettings {
+  def fromSubConfig(c: Config) = apply (
+    c.getLong("reservoir-highest-trackable-value"),
```

Comment:
What happens when one value is `null`? Should we validate first? Ah, you do so in the constructor...